portal-api/src/v1/roles/validate-user-is-acting-on-self-to-change-password-or-is-admin.js

```diff
@@ -0,0 +1,26 @@
+const { HttpStatusCode } = require('axios');
+const CONSTANT = require('../../constants');
+/**
+ * Creates a middleware that returns a 401 error if the current user logs as admin or other user, and calls the next handler otherwise.
+ * @param {{ allowedRoles: string[] }}
+ * @returns {(req, res, next) => void}
+ */
+const validateUserPermission = (req, res, next) => {
+  if (!req.user) {
+    return res.status(HttpStatusCode.Unauthorized).send('Unauthorized');
+  }
+
+  const isActingOnSelf = req?.user?._id === req?.params?._id;
+  const isAdmin = req?.user?.roles?.includes(CONSTANT.ADMIN);
+  const isMaker = req?.user?.roles?.includes(CONSTANT.MAKER);
+
+  if (!isActingOnSelf && !isAdmin && !isMaker) {
+    return res.status(HttpStatusCode.Forbidden).send('Forbidden');
+  }
+
+  return next();
+};
+
+module.exports = {
+  validateUserPermission,
+};
```

Comment:
Do we have some unit tests for this?

Can we remove `to-change-password` from the file name as the middleware doesn't have any mentions of password functionality.

Afterwards would be good to rename the middleware const (`validateUserPermission`) to align with the file name, like the existing `userHasAtLeastOneAllowedRole` middleware.

Reply:
Added unit test and renamed middleware to` validate-user-is-acting-on-self-or-is-admin.js`. Also renamed `const { validateUserPermission } = require('./roles/validate-user-is-acting-on-self-or-is-admin');`